beta-src/src/components/map/components/WDBuildContainer.tsx

```diff
@@ -15,23 +14,30 @@ import { useAppDispatch, useAppSelector } from "../../../state/hooks";
 import WDBuildUnitButtons from "./WDBuildUnitButtons";
 import { TerritoryMeta } from "../../../state/interfaces/TerritoriesState";
 import TerritoryMap from "../../../data/map/variants/classic/TerritoryMap";
+import provincesMapData from "../../../data/map/ProvincesMapData";
 
 const WDBuildContainer: React.FC = function (): React.ReactElement {
   const dispatch = useAppDispatch();
-  const territoriesMeta = useAppSelector(gameTerritoriesMeta);
   const maps = useAppSelector(gameMaps);
 
   const build = (availableOrder, canBuild, toTerrID) => {
     console.log(
       `Dispatched a build ${canBuild} ${BuildUnitMap[canBuild]} ${toTerrID}`,
     );
+    let terrIDToBuildOn = toTerrID;
+    if (BuildUnitMap[canBuild] === "Build Army") {
```

Comment:
Previously there wasn't any handling of slots at all, as you can see in processMouseClick.

There was in Codazen's branch, but we ripped out part of that logic, and/or they handled it in a way that I didn't want to replicate (namely, hardcoding special case logic for STP and/or overlaying additional copies of what is now ProvinceMapData that don't get drawn but cause some separate coastal mouse receivers to be generated).

The way it works now is that it uses the exact same function that we have for moves and retreats as for builds to determine what coast you click on, then when you do it gives you army/fleet options on both coasts (and since it uses the same function, it only distinguishes 2 click locations, not 3, as there is no separate "center" location that it tries to tell apart). If you pick the fleet, the fleet is build on that coast, if you pick the army the army is built on the root territory.